### PR TITLE
Port to netcoreapp2.0

### PR DESCRIPTION
**Pull request by droyad**
TODO:

- [x] See if we need long file name support under windows and `netcoreapp2.0`
- [x] Determine if WORKAROUND_FOR_EMPTY_STRING_BUG is required still
- [x] Get Linux build agents working
- [x] Test it out

### CODE REVIEW

**Comment by droyad** [comment]
The WORKAROUND_FOR_EMPTY_STRING_BUG workaround causes the `ApplyDeltaFixture`, `FindPackageFixture` and `PackageDownloadFixture` to fail (https://build.octopushq.com/viewLog.html?buildId=252933&tab=buildResultsDiv&buildTypeId=OctopusDeploy_CalamariNetCore_DotnetTesting_BuildAndTestRhel72DotNet)

**Comment by droyad** [comment]
netcoreapp2.0 supports long filenames OOTB

**Comment by droyad** [comment]
WORKAROUND_FOR_EMPTY_STRING_BUG is still needed

File: source/Calamari/Integration/Processes/Semaphores/SystemSemaphoreManager.cs [hunk 1]
```diff
@@ -22,9 +22,15 @@ public SystemSemaphoreManager(ILog log, TimeSpan initialWaitBeforeShowingLogMess
             this.initialWaitBeforeShowingLogMessage = (int)initialWaitBeforeShowingLogMessage.TotalMilliseconds;
         }
 
-#if NET40
         public IDisposable Acquire(string name, string waitMessage)
         {
+            return CalamariEnvironment.IsRunningOnWindows
+                ? AcquireSemaphore(name, waitMessage)
+                : AcquireMutex(name, waitMessage);
```
Comment (MJRichardson) [inline source/Calamari/Integration/Processes/Semaphores/SystemSemaphoreManager.cs]: Why can we not just use the Mutex on both Windows and Linux?  It's global right?

Comment (droyad) [inline source/Calamari/Integration/Processes/Semaphores/SystemSemaphoreManager.cs]: Yeh that would be good (this sparked my #feature-discussion thread). I think the initial approach was not to change functionality in the existing code, but this sounds like a good change.

Comment (droyad) [inline source/Calamari/Integration/Processes/Semaphores/SystemSemaphoreManager.cs]: Thinking about it, we should do it as a separate change and add a release note for it

**Pull request merged**